pkg/usagestats/stats.go

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package usagestats
+
+import (
+	"expvar"
+	"fmt"
+)
+
+const (
+	statsPrefix = "github.com/grafana/mimir/"
+	targetKey   = "target"
+	editionKey  = "edition"
+
+	EditionOSS        = "oss"
+	EditionEnterprise = "enterprise"
+)
+
+func init() {
+	SetEdition(EditionOSS)
+}
+
+// SetTarget sets the target name.
+func SetTarget(target string) {
+	NewString(targetKey).Set(target)
+}
+
+// SetEdition sets the edition name.
+func SetEdition(edition string) {
+	NewString(editionKey).Set(edition)
+}
+
+// NewString returns a new String stats object.
+// If a String stats object with the same name already exists it is returned.
+func NewString(name string) *expvar.String {
+	existing := expvar.Get(statsPrefix + name)
+	if existing != nil {
+		if s, ok := existing.(*expvar.String); ok {
+			return s
+		}
```

Comment:
This is a `NewString` function but it returns an existing string. I think this can hide undesired behaviors. Why would we expect this to overwrite a string?

Reply:
It was copied from Loki, and I just kept logic as is. The way this stat function (and functions that will be introduced in the next PR) are intended to use, is just as a "get or create". I can change the name function to make it more clear.

Reply:
I'd rename & unexport then.